tests/testthat/test-get_source_expressions.R

```diff
@@ -122,3 +122,23 @@ test_that("Warns if encoding is misspecified", {
   expect_equal(the_lint$message, "Invalid multibyte string. Is the encoding correct?")
   expect_equal(the_lint$line_number, 1L)
 })
+
+test_that("Can extract line number from parser errors", {
+  skip_if_not(getRversion() >= "4.0")
+
+  with_content_to_parse("\"ok\"\nR\"---a---\"", {
```

Comment:
the regex has three cases:

or(" at ", " on ", " ("),

can we make sure we have tests of each? maybe add a comment to the test saying which is tested

Reply:
Good point. Added a missing test case (on line x is only triggered by a repeated formal argument)